pkg/build/api/validation/validation.go

```diff
@@ -29,6 +29,32 @@ func ValidateBuild(build *buildapi.Build) field.ErrorList {
 	allErrs := field.ErrorList{}
 	allErrs = append(allErrs, validation.ValidateObjectMeta(&build.ObjectMeta, true, validation.NameIsDNSSubdomain, field.NewPath("metadata"))...)
 	allErrs = append(allErrs, validateCommonSpec(&build.Spec.CommonSpec, field.NewPath("spec"))...)
+
+	// Check that all source image references are of DockerImage kind.
+	strategyPath := field.NewPath("spec", "strategy")
+	switch {
+	case build.Spec.Strategy.SourceStrategy != nil:
+		allErrs = append(allErrs, validateBuildImageReference(&build.Spec.Strategy.SourceStrategy.From, strategyPath.Child("sourceStrategy").Child("from"))...)
+		allErrs = append(allErrs, validateBuildImageReference(build.Spec.Strategy.SourceStrategy.RuntimeImage, strategyPath.Child("sourceStrategy").Child("runtimeImage"))...)
+	case build.Spec.Strategy.DockerStrategy != nil:
+		allErrs = append(allErrs, validateBuildImageReference(build.Spec.Strategy.DockerStrategy.From, strategyPath.Child("dockerStrategy").Child("from"))...)
+	case build.Spec.Strategy.CustomStrategy != nil:
+		allErrs = append(allErrs, validateBuildImageReference(&build.Spec.Strategy.CustomStrategy.From, strategyPath.Child("customStrategy").Child("from"))...)
```

Comment:
My understanding is that custom strategy is implemented by means of docker image, the one `from` points to, so I'd say no. IOW the field is useful for build controller, not for the strategy itself.

The `from` field of custom strategy is used in this function: https://github.com/openshift/origin/blob/v1.4.0-alpha.0/pkg/build/controller/strategy/custom.go#L24 You can notice that the `kind` is not checked.